src/ray/core_worker/experimental_mutable_object_manager.cc

```diff
@@ -159,230 +172,191 @@ void ExperimentalMutableObjectManager::DestroySemaphores(const ObjectID &object_
   semaphores_.erase(object_id);
 }
 
-Status ExperimentalMutableObjectManager::WriteAcquire(const ObjectID &object_id,
-                                                      int64_t data_size,
-                                                      const uint8_t *metadata,
-                                                      int64_t metadata_size,
-                                                      int64_t num_readers,
-                                                      std::shared_ptr<Buffer> *data) {
-  auto writer_channel_entry = writer_channels_.find(object_id);
-  if (writer_channel_entry == writer_channels_.end()) {
+Status MutableObjectManager::WriteAcquire(const ObjectID &object_id,
+                                          int64_t data_size,
+                                          const uint8_t *metadata,
+                                          int64_t metadata_size,
+                                          int64_t num_readers,
+                                          std::shared_ptr<Buffer> &data) {
+  WriterChannel *channel = GetChannel<WriterChannel>(writer_channels_, object_id);
+  if (!channel) {
     return Status::ObjectNotFound("Writer channel has not been registered");
   }
+  RAY_CHECK(!channel->written);
```

Comment:
What's the reasoning for changing this from returning a status to asserting?

Reply:
Discussed offline. Done.